### PR TITLE
Demo of chapter 19 (4th eidtion)

### PR DESCRIPTION
**Pull request by bruceyoungsysu**
- learners
- Loss functions and layers
- Optimizer and backpropagation
- RNN

### CODE REVIEW

**Comment by delaray** [comment]
Very good @bruceyoungsysu !
